actions/image.search.by.property.yaml

```diff
@@ -0,0 +1,144 @@
+    default: workflows.search_by_property.search_by_property
+    immutable: true
+    type: string
+  query_type:
+    default: ImageQuery
+    immutable: true
+    type: string
+  as_admin:
+    default: True
+    immutable: true
+    type: boolean
+  cloud_account:
+    description: The clouds.yaml account to use whilst performing this action
+    required: true
+    type: string
+    default: "dev"
+    enum:
+      - "dev"
+      - "prod"
+  properties_to_select:
+    default:
+      - "image_creation_date"
+      - "image_creation_progress"
+      - "image_id"
+      - "image_last_updated_date"
+      - "image_minimum_ram"
+      - "image_minimum_disk"
+      - "image_name"
+      - "image_size"
+      - "image_status"
+    type: array
+    description: "
+    A comma-spaced list of image properties to display for the resulting images - leave empty for all properties. One of:
+      - 'image_creation_date'
+      - 'image_creation_progress'
+      - 'image_id'
+      - 'image_last_updated_date'
+      - 'image_minimum_ram'
+      - 'image_minimum_disk'
+      - 'image_name'
+      - 'image_size'
+      - 'image_status'
+    Anything else will raise an error"
+    required: false
+  output_type:
+    default: "to_string"
+    type: string
+    enum:
+      - "to_html"
+      - "to_objects"
+      - "to_props"
+      - "to_string"
+    description: "
+    A string representing how to return the results of the query
+      - 'to_html' - a tabulate table (in html)
+      - 'to_props' - properties dicts as a python list
+      - 'to_objects - as a list of openstack resources
+      - 'to_string' - a tabulate table"
+    required: true
+  property_to_search_by:
+    default: "image_name"
+    description: "choose property to search by (acts as OR for each)"
+    enum:
+      - "image_creation_date"
+      - "image_creation_progress"
+      - "image_id"
+      - "image_last_updated_date"
+      - "image_minimum_ram"
+      - "image_minimum_disk"
+      - "image_name"
+      - "image_size"
+      - "image_status"
+    type: string
+    required: true
+  search_mode:
+    description: "what to mode to use to search by"
+    default: "any_in"
+    type: string
+    required: true
+    enum:
+      - "any_in"
+      - "not_any_in"
+  values:
+    description: "a comma-spaced list of values to search for"
+    required: true
+    type: array
+  from_projects:
+    description: "(Optional) comma-spaced list of projects id/names to limit query to, if not provided, runs against all projects"
+    type: array
+    default: null
+    required: false
+  all_projects:
+    type: boolean
+    description: "tick to search in all projects - default True"
+    required: true
+    default: true
+  group_by:
+    description: "(Optional) image property to group unique results by"
+    type: string
+    required: false
+    default: null
+    enum:
+      - null
+      - "image_creation_date"
+      - "image_creation_progress"
+      - "image_id"
```

Comment:
I don't think grouping by image_id is needed

Reply:
have removed and also remove image_name for the same reason from all of them